Adds a management command to create and optionally enroll a user

jkachel opened the pull request:
#### Pre-Flight checklist

- [X] Testing
  - [X] Code is tested
  - [X] Changes have been manually tested

#### What are the relevant tickets?

Part 1 of #1152 

#### What's this PR do?

This adds a management command that will create and optionally enroll a new user in the system. The enrollment is done using the existing `create_enrollment` command.

#### How should this be manually tested?

Run the management command: 
`./manage.py create_user <username> <email> <firstname> <lastname> <displayname> <countrycode> [--enroll <courseware>]`

This should create the user and, if you've supplied one or more `--enroll` flags, enroll them in the specified courseware object (either a course run or a program run).

rachellougee reviewed (changes requested):
The overall functionality works ok. There are a couple of small things for improvement if it's easy to implement
1. Got stack trace errors on 'duplicate key value violates unique constraint "users_user_username_key"', could we print out a readable message if a duplicate is found?
2. got "TypeError: object of type 'NoneType' has no len()" when running the command without --enroll argument
3. got stack trace errors on 'openedx.exceptions.NoEdxApiAuthError', this will happen whenever new user is created since this new user isn't auth in edx, could we create auth token if it's easy to do? if not, maybe just print out a readable message?

jkachel commented:
Addressed comments in https://github.com/mitodl/mitxonline/pull/1168/commits/965dc5caf72beea37cefbe9029bd6520bd601136 - 1 and 2 I resolved, but 3 I don't know that I can do a whole lot about. The underlying `create_run_enrollments` call logs that particular exception and that's what you're actually seeing there. The enrollment persists on the MITxOnline side (since I'm specifying `keep_failed_enrollments=True`) but not sure I want to change the actual enrollment code for this.

rachellougee reviewed (approved):
LGTM. didn't notice keep_failed_enrollments flag but agree no need to touch enrollment code
